Zed crashes when opening remote server

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Open remote server. Select the server. Zed crashes. And, I can no longer open that remote no matter what I do. This did work great until a few hours ago; I was able to edit my remotes with no issues.

I've deleted the `.zed_server` didrectory and `.local/share/zed` to hopefully reinitialize,but zed will no longer connect to my remote.

I've also tried to kill the running language servers, and see if zed can reboot them, but no luck:

```
chrisdawson@lima-tailscale-extrastatic:~$ ps auwx | grep zed
chrisda+   32100  0.1  0.9  52640 18932 ?        Sl   13:56   0:00 /home/chrisdawson.linux/.zed_server/zed-remote-server-stable-linux-aarch64 run --log-file /home/chrisdawson.linux/.local/share/zed/logs/server-workspace-32.log --pid-file /home/chrisdawson.linux/.local/share/zed/server_state/workspace-32/server.pid --stdin-socket /home/chrisdawson.linux/.local/share/zed/server_state/workspace-32/stdin.sock --stdout-socket /home/chrisdawson.linux/.local/share/zed/server_state/workspace-32/stdout.sock --stderr-socket /home/chrisdawson.linux/.local/share/zed/server_state/workspace-32/stderr.sock
chrisda+   32276  3.0  5.7 669908 114700 ?       Sl   13:56   0:01 /usr/bin/node /home/chrisdawson.linux/.local/share/zed/languages/pyright/node_modules/pyright/langserver.index.js --stdio
chrisda+   32333  0.0  0.0   6672  1792 pts/1    S+   13:57   0:00 grep --color=auto zed
chrisdawson@lima-tailscale-extrastatic:~$ kill -9 32100 32276
```

### Environment

Zed: v0.159.7 (Zed)
OS: macOS 13.6.7
Memory: 16 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>
<!-- Click below this line and paste or drag-and-drop your log-->
{"thread":"main","payload":"called create_local_buffer on a remote project","location_data":{"file":"crates/project/src/project.rs","line":1740},"backtrace":["backtrace::capture::Backtrace::create","backtrace::capture::Backtrace::new","zed::reliability::init_panic_hook::{{closure}}","std::panicking::rust_panic_with_hook","std::panicking::begin_panic_handler::{{closure}}","std::sys::backtrace::__rust_end_short_backtrace","_rust_begin_unwind","core::panicking::panic_fmt","project::Project::create_local_buffer","editor::items::<impl workspace::item::SerializableItem for editor::Editor>::deserialize::{{closure}}::{{closure}}","async_task::raw::RawTask<F,T,S,M>::run","gpui::platform::mac::dispatcher::trampoline","<gpui::platform::mac::platform::MacPlatform as gpui::platform::Platform>::run","gpui::app::App::run","zed::main","std::sys::backtrace::__rust_begin_short_backtrace","std::rt::lang_start::{{closure}}","std::rt::lang_start_internal","_main"],"app_version":"0.159.7","release_channel":"Zed","os_name":"macOS","os_version":"13.6.7","architecture":"aarch64","panicked_on":1730659907427,"system_id":"ade1e066-1bb0-407c-95ff-2ac78ff38e23","installation_id":"0a811b75-8626-419b-b143-acad02ef517f","session_id":"78b4fa2c-d917-471a-b8ba-1495c245e58a"}

<!-- Click above this line and paste or drag-and-drop your log--></pre></details>

Full stack trace.

[zed_stacktrace.log](https://github.com/user-attachments/files/17611738/zed_stacktrace.log)

Forgot zed.log

```
2024-11-03T14:29:09.957523-05:00 [INFO] ========== starting zed ==========
2024-11-03T14:29:09.96114-05:00 [INFO] Opening main db
2024-11-03T14:29:09.964904-05:00 [INFO] Opening main db
2024-11-03T14:29:10.002311-05:00 [INFO] Using git binary path: Some("/Applications/Zed.app/Contents/MacOS/git")
2024-11-03T14:29:10.234015-05:00 [INFO] no tokio runtime found, creating one for Reqwest...
2024-11-03T14:29:10.264098-05:00 [INFO] initializing context server client
2024-11-03T14:29:10.264447-05:00 [INFO] Registering built-in prompt template: content_prompt
2024-11-03T14:29:10.265869-05:00 [INFO] Registering built-in prompt template: edit_workflow
2024-11-03T14:29:10.266045-05:00 [INFO] Registering built-in prompt template: project_slash_command
2024-11-03T14:29:10.266133-05:00 [INFO] Registering built-in prompt template: terminal_assistant_prompt
2024-11-03T14:29:10.266479-05:00 [INFO] No prompt template overrides directory found at /Users/chrisdawson/.config/zed/prompt_overrides. Using built-in prompts.
2024-11-03T14:29:10.268292-05:00 [INFO] extensions updated. loading 4, reloading 0, unloading 0
2024-11-03T14:29:10.339744-05:00 [INFO] Opening main db
2024-11-03T14:29:10.410515-05:00 [INFO] set status on client 0: Authenticating
2024-11-03T14:29:10.428643-05:00 [INFO] Opening main db
2024-11-03T14:29:10.433052-05:00 [INFO] set status on client 225684: Connecting
2024-11-03T14:29:10.44214-05:00 [INFO] Opening main db
2024-11-03T14:29:10.530265-05:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-11-03T14:29:10.559487-05:00 [INFO] set environment variables from shell:/bin/zsh, path:/Users/chrisdawson/.cargo/bin:/Users/chrisdawson/.nvm/versions/node/v20.17.0/bin:/Users/chrisdawson/Library/pnpm:/Users/chrisdawson/Downloads/google-cloud-sdk/bin:/Users/chrisdawson/perl5/bin:/Users/chrisdawson/miniconda3/bin:/Library/Frameworks/Python.framework/Versions/3.12/bin:/Users/chrisdawson/.wasmedge/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/go/bin:/Library/Apple/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Users/chrisdawson/go/bin/:/Users/chrisdawson/bin/:/Applications/Docker.app/Contents/Resources/bin/
2024-11-03T14:29:10.916664-05:00 [INFO] add connection to peer
2024-11-03T14:29:10.916751-05:00 [INFO] add_connection;
2024-11-03T14:29:10.916828-05:00 [INFO] waiting for server hello
2024-11-03T14:29:10.928506-05:00 [INFO] got server hello
2024-11-03T14:29:10.929024-05:00 [INFO] set status to connected (connection id: ConnectionId { owner_id: 0, id: 0 }, peer id: PeerId { owner_id: 606, id: 3118513 })
2024-11-03T14:29:10.929328-05:00 [INFO] set status on client 225684: Connected { peer_id: PeerId { owner_id: 606, id: 3118513 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2024-11-03T14:29:11.039993-05:00 [INFO] starting language server process. binary path: "/Users/chrisdawson/.nvm/versions/node/v20.17.0/bin/node", working directory: "/", args: ["/Users/chrisdawson/Library/Application Support/Zed/copilot/copilot-v0.5.0/dist/agent.js", "--stdio"]
2024-11-03T14:29:11.103835-05:00 [ERROR] cannot read LSP message headers
2024-11-03T14:29:11.103963-05:00 [ERROR] Broken pipe (os error 32)
2024-11-03T14:29:11.104061-05:00 [ERROR] server shut down
2024-11-03T14:29:33.799032-05:00 [INFO] server binary is opened by another process. not updating
2024-11-03T14:29:33.846182-05:00 [ERROR] client:unhandled ssh message name:UpdateProject
2024-11-03T14:29:34.085581-05:00 [ERROR] {
  "thread": "main",
  "payload": "called create_local_buffer on a remote project",
  "location_data": {
    "file": "crates/project/src/project.rs",
    "line": 1740
  },
  "backtrace": [
    "backtrace::capture::Backtrace::create",
    "backtrace::capture::Backtrace::new",
    "zed::reliability::init_panic_hook::{{closure}}",
    "std::panicking::rust_panic_with_hook",
    "std::panicking::begin_panic_handler::{{closure}}",
    "std::sys::backtrace::__rust_end_short_backtrace",
    "_rust_begin_unwind",
    "core::panicking::panic_fmt",
    "project::Project::create_local_buffer",
    "editor::items::<impl workspace::item::SerializableItem for editor::Editor>::deserialize::{{closure}}::{{closure}}",
    "async_task::raw::RawTask<F,T,S,M>::run",
    "gpui::platform::mac::dispatcher::trampoline",
    "<gpui::platform::mac::platform::MacPlatform as gpui::platform::Platform>::run",
    "gpui::app::App::run",
    "zed::main",
    "std::sys::backtrace::__rust_begin_short_backtrace",
    "std::rt::lang_start::{{closure}}",
    "std::rt::lang_start_internal",
    "_main"
  ],
  "app_version": "0.159.7",
  "release_channel": "Zed",
  "os_name": "macOS",
  "os_version": "13.6.7",
  "architecture": "aarch64",
  "panicked_on": 1730662174085,
  "system_id": "ade1e066-1bb0-407c-95ff-2ac78ff38e23",
  "installation_id": "0a811b75-8626-419b-b143-acad02ef517f",
  "session_id": "3d93d205-627d-41bd-8ba3-727166f6adc9"
}

```

--- a/crates/editor/src/items.rs
+++ b/crates/editor/src/items.rs
@@ -992,12 +992,15 @@ impl SerializableItem for Editor {
                     };
 
                     // First create the empty buffer
-                    let buffer = project.update(&mut cx, |project, cx| {
-                        project.create_local_buffer("", language, cx)
-                    })?;
+                    let buffer = project
+                        .update(&mut cx, |project, cx| project.create_buffer(cx))?
+                        .await?;
 
                     // Then set the text so that the dirty bit is set correctly
                     buffer.update(&mut cx, |buffer, cx| {
+                        if let Some(language) = language {
+                            buffer.set_language(Some(language), cx);
+                        }
                         buffer.set_text(contents, cx);
                     })?;
 

